### PR TITLE
Support more searchable file types in Elasticsearch

### LINKED ISSUE #14154: Add more searchable media types in Elasticsearch/Lucene
### Is your feature request related to a problem? Please describe.
Currently we only support searching in PDF files in Elasticsearch.

### Describe the solution you'd like
We should be able to search .doc, .docx, .txt, .rtf, .ppt, .pptx. We can easily use  Spire.Doc to add this suppor

### Describe alternatives you've considered
I added support to all these in my own project

### PR DESCRIPTION
**Pull request by MikeAlhayek**
Fix #14154

### CODE REVIEW

**Review by sebastienros — APPROVED** [review]
If we could have different features for the document types
- pdf
- txt
- office

**Review by sebastienros — COMMENTED** [review]
And maybe find a better option to convert the files, or at least only support docx/pptx which should be common compared to doc/ppt

**Comment by MikeAlhayek** [comment]
@sebastienros I added a new feature called "Media Indexer" which allows the user o index more media file. I can't find a better library than the one I used here `FreeSpire.Doc`.

If someone can suggest a better library, we can always change the implementations.

**Comment by giannik** [comment]
> @sebastienros I added a new feature called "Media Indexer" which allows the user o index more media file. I can't find a better library than the one I used here `FreeSpire.Doc`.
> 
> If someone can suggest a better library, we can always change the implementations.

I have used this library successfully, totally open source: 
 DocumentFormat.OpenXml Version="2.20.0"
 
 and example : 
 
 
 ```

using System;
using System.Collections.Generic;
using System.IO;
using System.Text;
using System.Threading.Tasks;
using DocumentFormat.OpenXml.Packaging;
using DocumentFormat.OpenXml.Wordprocessing;

namespace OrchardCore.Media.Indexing
{
    public class DocxMediaFileTextProvider : IMediaFileTextProvider
    {
        public   Task<string> GetTextAsync(string path, Stream fileStream)
        {

            var wordprocessingDocument = WordprocessingDocument.Open(fileStream, false);
            try
            {
                StringBuilder sb = new();

                MainDocumentPart mainPart = wordprocessingDocument.MainDocumentPart;
                if (mainPart is null)
                {
                    throw new InvalidOperationException("The main document part is missing.");
                }

                Body body = mainPart.Document.Body;
                if (body is null)
                {
                    throw new InvalidOperationException("The document body is missing.");
                }

                IEnumerable<Paragraph> paragraphs = body.Descendants<Paragraph>();
                if (paragraphs != null)
                {
                    foreach (Paragraph p in paragraphs)
                    {
                        sb.AppendLine(p.InnerText);
                    }
                }

                return Task.FromResult(sb.ToString());
            }
            finally
            {
                wordprocessingDocument.Dispose();
            }

        }
    }

```

**Comment by MikeAlhayek** [comment]
@giannik thank you. Does it allow reading other types like pptx and .rtf?

**Comment by giannik** [comment]
> @giannik thank you. I think it also supports pptx. I'll look more at it tomorrow.

oh yes,Word, Excel, and PowerPoint. the  sample code is just an example

**Comment by MikeAlhayek** [comment]
@giannik I replaced the libraries. Thanks for sharing :)

**Comment by giannik** [comment]
@MikeAlhayek . thank you for the work. maybe an improvement would be to replace StringBuilder with ZString.CreateStringBuilder();

**Comment by MikeAlhayek** [comment]
> @MikeAlhayek . thank you for the work. maybe an improvement would be to replace StringBuilder with ZString.CreateStringBuilder();

Yes. Good idea. Updated!

File: src/OrchardCore.Modules/OrchardCore.Media/Manifest.cs [hunk 1]
```diff
@@ -38,8 +38,8 @@
 )]
 
 [assembly: Feature(
-    Id = "OrchardCore.Media.Indexer",
-    Name = "Media Indexer",
+    Id = "OrchardCore.Media.Indexing",
+    Name = "Media Indexing",
     Description = "Providers a way to index media file with common formats in Lucene and Elasticsearch.",
```
[the remark this answers is not in the document]
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Media/Manifest.cs:43]: opps, i'll fix it.

**Pull request merged**